File: Basic/Does array represent Heap.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends

class Solution{
    public:
    bool isMaxHeap(int a[], int n)
    {
        // Your code goes here
        
        int nodo1 = 0;
        int nodo2 = 0;
        
        for (int i=1; i<n; i++)
        {
            if (nodo2==2) nodo1++;
            
            if (a[nodo1] > a[i]) nodo2++;
            else return false;
        }
        
        return true;
    }
};

// { Driver Code Starts.
int main() {
    
    int t;
    cin >> t;
    while(t--)
    {
       int n;
       cin >> n;
       int a[4*n]={0};
       for(int i =0;i<n;i++){
           cin >> a[i];
       }
       Solution ob;
       cout<<ob.isMaxHeap(a, n)<<endl;
        
    }
    return 0;
}
  // } Driver Code Ends

File: Easy/Does array represent Heap.py

class Solution:
    def isMaxHeap(self, arr, n):
        # Your code goes here
        for i in range(n):
            left = (i * 2) + 1
            right = left + 1
            
            if (left < n and arr[i] < arr[left]) or (right < n and arr[i] < arr[right]):
                return False
            

        return True

#{ 
 # Driver Code Starts
if __name__ =='__main__':
    t= int(input())
    for tcs in range(t):
        n=int(input())
        arr=[int(x) for x in input().split()]
        ob=Solution()
        print(int(ob.isMaxHeap(arr,n)))
# } Driver Code Ends